src/DataService/NotificationTypeExtractor.php

```diff
@@ -30,7 +30,7 @@ public function setClass(string $class)
     /**
      * {@inheritdoc}
      */
-    public function getTypes($class, $property, array $context = [])
+    public function getTypes($class, $property, array $context = []): ?array
```

Comment:
This technically represents an API break. In retrospect, I think we should have marked some classes like this `@internal`.

Given we don't know of a single external use of this library, and that I'm guessing that returning a non-array would break things anyways, I think we can document this in the changelog instead of forcing a whole new major version.